### PR TITLE
Adapting Muon to AdaGC

### PR DESCRIPTION
**Pull request by xxyux**
### PR Category

Execute Infrastructure

### PR Types

New features

### Description

This PR will integrate [AdaGC: Enhancing LLM Pretraining Stability via Adaptive Gradient Clipping](https://arxiv.org/abs/2502.11034) into the framework, and adaptes the Muon optimizer to AdaGC.

The specific implementation of AdaGC is located at: `Paddle/python/paddle/distributed/fleet/meta_optimizers/dygraph_optimizer/hybrid_parallel_optimizer.py.` 
The code for adapting the Muon optimizer is at: `Paddle/python/paddle/distributed/fleet/meta_optimizers/muon_sharding_optimizer.py`. 

Finally, the unit tests for the Muon optimizer have been modified accordingly.

Co-Authored with @GuoxiaWang 

### 是否引起精度变化

否

### CODE REVIEW

**Review by GuoxiaWang — DISMISSED** [review]

**Review by sneaxiy — APPROVED** [review]
LGTM
